package.json

```diff
@@ -11,12 +11,22 @@
   "homepage": "https://github.com/okta/samples-nodejs-express-4",
   "scripts": {
     "postinstall": "node post-install.js",
-    "custom-login": "node custom-login/server.js",
-    "okta-hosted-login": "node okta-hosted-login/server.js",
+    "custom-login-server": "node custom-login/server.js",
+    "test:custom-login": "protractor okta-oidc-tck/e2e-tests/custom-login/conf.js",
+    "okta-hosted-login-server": "node okta-hosted-login/server.js",
+    "test:okta-hosted-login": "protractor okta-oidc-tck/e2e-tests/okta-hosted-login/conf.js",
     "resource-server": "node resource-server/server.js",
-    "test": "echo \"Tests not yet implemented\""
+    "test": "npm run test:e2e",
+    "test:e2e": "npm run get-oidc-tck && npm run test:okta-hosted-login && npm run test:custom-login",
+    "get-oidc-tck": "[ ! -d okta-oidc-tck/ ] && git clone https://github.com/okta/okta-oidc-tck.git",
```

Comment:
If there is no else conditional this doesn't work the way we expect because the non-zero exit falls through to the npm runner.  Here is one solution:

```bash
[ ! -d okta-oidc-tck/ ] && git clone https://github.com/okta/okta-oidc-tck.git || echo "TCK already cloned"
```

Reply:
Done